### PR TITLE
PRP-351 Update database module to use RDS instead of Aurora

### PR DESCRIPTION
**Pull request by aplybeah**
## Ticket

https://wicmtdp.atlassian.net/browse/PRP-351

## Changes
* Added a new database child module `database-rds`
* moved all database associations to the new module

## Testing
1. Log into the AWS console and find the `dev` database in the `RDS` section of the database console
2. Visit the dev portal and ensure that all activity is the same

### CODE REVIEW

File: infra/app/env-template/main.tf [hunk 1]
```diff
@@ -246,7 +246,7 @@ module "staff" {
 }
 
 module "analytics_database" {
-  source        = "../../modules/database"
+  source        = "../../modules/database-rds"
```
Comment (microwavenby) [inline infra/app/env-template/main.tf]: I think this naming may be unclear at some point in the future, as Aurora is a valid RDS target as well as serverless. 

Maybe database-serverless instead?

File: infra/modules/database-serverless/README.md
Comment (rocketnova) [inline infra/modules/database-serverless/README.md:1]: I've been removing the README files from inside individual terraform modules and putting inline documentation in the `main.tf` instead. Let's follow that model for this module too.

File: infra/modules/database-serverless/main.tf [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  }
+}
+
+############################
+## Database Configuration ##
+############################
+resource "aws_db_instance" "database" {
+  # checkov:skip=CKV_AWS_157:Multi-AZ is mostly unessecary for a project of this size.
+  identifier        = var.database_name
+  allocated_storage = 20
```
Comment (rocketnova) [inline infra/modules/database-serverless/main.tf]: Please verify:
- that this is the default recommended storage 
- AND that it's sufficient for both of our applications

File: infra/modules/database-serverless/main.tf [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  # "postgresql isn't a supported engine name in RDS. use postgres instead"
```
Comment (rocketnova) [inline infra/modules/database-serverless/main.tf]:
```suggestion
```

File: infra/modules/database-serverless/main.tf [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+  engine                              = var.database_type == "mysql" ? "mysql" : "postgres"
+  engine_version                      = var.database_type == "mysql" ? "8.0" : "14.6"
+  instance_class                      = "db.t3.micro"
+  db_name                             = local.database_name_formatted
+  port                                = var.database_port
+  enabled_cloudwatch_logs_exports     = [var.database_type == "mysql" ? "general" : "postgresql"]
+  apply_immediately                   = true
+  deletion_protection                 = true
+  storage_encrypted                   = true
+  skip_final_snapshot                 = true
```
Comment (rocketnova) [inline infra/modules/database-serverless/main.tf]:
```suggestion
  skip_final_snapshot                 = false
```

File: infra/modules/database-serverless/main.tf
Comment (rocketnova) [inline infra/modules/database-serverless/main.tf:1]: Add in a backup policy. Does not need to use the AWS Backup service, but we do want to have configuration for being able to regularly back up the database and be able to restore from those backups.

File: infra/modules/database/outputs.tf
Comment (rocketnova) [inline infra/modules/database/outputs.tf:1]: @aplybeah Why are we using the `secret_name`s here and not the `secret.arn`s as in https://github.com/navapbc/wic-participant-recertification-portal-demo/blob/main/infra/modules/database/outputs.tf?

Comment (rocketnova) [reply infra/modules/database/outputs.tf:1]: I think this is a mistake. I think the the `env-template/main.tf` expects secret arns

Comment (aplybeah) [reply infra/modules/database/outputs.tf:1]: ah, great catch. That was a copy/paste mistake on my part

**Review by rocketnova — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**